cfgmgr/coppmgr.cpp

```diff
@@ -610,21 +610,18 @@ void CoppMgr::doCoppTrapTask(Consumer &consumer)
                 (trap_group == m_coppTrapConfMap[key].trap_group) &&
                 (trap_ids == m_coppTrapConfMap[key].trap_ids))
             {
-                std::vector<FieldValueTuple> feature_fvs;
+                m_coppTrapConfMap[key].is_always_enabled = is_always_enabled;
                 if (is_always_enabled == "true")
                 {
                     /* if the value was changed from false to true,
                     if the trap is not installed, install it.
                     otherwise, do nothing. */
 
-                    if (m_coppTrapConfMap.find(key) == m_coppTrapConfMap.end())
+                    // if the feature was not enabled, install the trap
```

Comment:
I think feature enabled should be checked here before the cache is updated since feature enabled looks at the cache and the point of the logic is to get the feature state before the always enabled update. Can you please revisit?